src/legacy/ui/public/agg_types/agg_params.ts

```diff
@@ -17,46 +17,43 @@
  * under the License.
  */
 
-import { IndexedArray } from '../indexed_array';
+import { AggParamType } from 'ui/agg_types/param_types/agg';
 import { FieldParamType } from './param_types/field';
 import { OptionedParamType } from './param_types/optioned';
 import { StringParamType } from './param_types/string';
 import { JsonParamType } from './param_types/json';
 import { BaseParamType } from './param_types/base';
-import { createLegacyClass } from '../utils/legacy_class';
+import { AggConfig } from '../vis/agg_config';
+import { AggConfigs } from '../vis/agg_configs';
+
+export type AggParam = BaseParamType;
+
+export interface AggParamOption {
+  val: string;
+  display: string;
+  enabled?(agg: AggConfig): void;
+}
 
 const paramTypeMap = {
   field: FieldParamType,
-  optioned: (OptionedParamType),
-  string: (StringParamType),
-  json: (JsonParamType),
-  _default: (BaseParamType)
-};
+  optioned: OptionedParamType,
+  string: StringParamType,
+  json: JsonParamType,
+  agg: AggParamType,
+  _default: BaseParamType,
+} as Record<string, any>;
```

Comment:
I don't know if `as Record<string, any>` is needed here -- since it is an object literal, doesn't TS already infer the types?

This might also make it so you can remove the `as AggParam[]` casting on L54 below, which would be nice since it feels like that casting shouldn't be needed anyway (haven't tested this, I'm just speculating based on reading the code)

Reply:
without it i get `Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{ field: typeof FieldParamType; optioned: typeof OptionedParamType; string: typeof StringParamType; json: typeof JsonParamType; agg: typeof AggParamType; _default: typeof BaseParamType; }'.`